sdk/storage/storage-blob/test/blobclient.spec.ts

```diff
@@ -51,6 +51,88 @@ describe("BlobClient", () => {
     }
   });
 
+  it.only("Set blob tags should work", async () => {
+    const tags = {
+      tag1: "val1",
+      tag2: "val2"
+    };
+    await blockBlobClient.setTags(tags);
+
+    const response = await blockBlobClient.getTags();
+    assert.deepStrictEqual(response.tags, tags);
+
+    const properties = await blockBlobClient.getProperties();
+    assert.deepStrictEqual(properties.tagCount, 2);
+
+    const download = await blockBlobClient.download();
+    assert.deepStrictEqual(download.tagCount, 2);
+
+    const listblob = containerClient.listBlobsFlat({ includeTags: true });
+
+    const iter = listblob.byPage();
+    const segment = await iter.next();
+
+    // TODO: Make blob tag type consistency cross all request or response
+    assert.deepStrictEqual(segment.value.segment.blobItems[0].tags, tags);
+  });
+
+  it.only("Get blob tags should work with a snapshot", async () => {
```

Comment:
has tag supported versioning?

Reply:
Yes. versionid is in client url. So no additional change for tags related methods.